[docs] Add a reference to the Redux DevTools expo devtools plugin (#30259)
expo/expo: docs/pages/debugging/devtools-plugins.mdx

In the terminal, run `npx expo start`, press <kbd>shift</kbd> + <kbd>m</kbd> to open the list of dev tools, and then select the React Query plugin. This will open the plugin's web interface, displaying queries as they are used in your app.

> ### Redux DevTools

Reviewer: ```suggestion
### Redux
```

Let's just call it Redux since it appears Redux DevTools is the name of the actual tool and this a plugin. This is also the pattern we follow for other libraries on this page.

> Based on the full [Redux DevTools](https://github.com/reduxjs/redux-devtools/) (from the Chrome extension) but as an Expo Dev Plugin.
> 
> - Easy to install and run.
> - Live list of actions and how it affects the state.
> - Ability to rewind, replay, and dispatch actions from the devtools.

Reviewer: ```suggestion
The `redux-devtools-expo-dev-plugin` is based on the [Redux DevTools](https://github.com/reduxjs/redux-devtools/) (from the Chrome extension). It provides a live list of actions and how they affect the state, and the ability to rewind, replay, and dispatch actions from the DevTools.
```

I updated the this sub-section because I don't think list item no. 1 is worth mentioning and we can introduce the plugin in the first line as the main subject matter here. Then, we can call the features this plugin has within the same paragraph.

To use the plugin, start by installing the package:

<Terminal cmd={['$ npx expo install redux-devtools-expo-dev-plugin']} />

> If you're using `@reduxjs/toolkit` then you need to modify your call to `configureStore` to disable the built in devtools (by passing in `devTools: false`) and add in the Expo Devtools plugin enhancer (by concatenating the `devToolsEnhancer()`).
> 
> Your call to `configureStore` will end up looking like this:

Reviewer: ```suggestion
If you're using `@reduxjs/toolkit`, modify the `configureStore` call to disable the built-in dev tools by passing in `devTools: false`. Then, add in the Expo DevTools plugin enhancer by concatenating the `devToolsEnhancer()`. The `configureStore` call is going to look like the following:
```

Let's also update the above paragraph. Instead of using brackets, since those user-facing changes, let's make part of the main sentences and also split the sentences here for better readability.

> ```javascript store.js

Reviewer: ````suggestion
```js store.js
````

Let's use the code language shorthand here, which is what we use for other code blocks for consistency.

import devToolsEnhancer from 'redux-devtools-expo-dev-plugin';

const store = configureStore({
  reducer: rootReducer,
  devTools: false,
  enhancers: getDefaultEnhancers => getDefaultEnhancers().concat(devToolsEnhancer()),
});
```

> In the terminal, run `npx expo start`, press <kbd>shift</kbd> + <kbd>m</kbd> to open the list of dev tools, and then select the Tinybase plugin. This will open the plugin's web interface, displaying the actions and contents of your store as actions are dispatched.

Reviewer: ```suggestion
In the terminal, run `npx expo start`, press <kbd>shift</kbd> + <kbd>m</kbd> to open the list of dev tools, and then select `redux-devtools-expo-dev-plugin`. This will open the plugin's web interface, displaying the actions and contents of your store as actions are dispatched.
```

Let's fix the typo by using this plugin's name when selecting in the CLI.

> For full installation instructions, including if your using `redux` directly rather than `@reduxjs/toolkit`, [see the project README](https://github.com/matt-oakes/redux-devtools-expo-dev-plugin).

Reviewer: ```suggestion
For complete installation and usage instructions, including if you're using `redux` directly rather than `@reduxjs/toolkit`, [see the project's README](https://github.com/matt-oakes/redux-devtools-expo-dev-plugin).
```

Since the readme of the plugin also include more than installation instructions, let's highlight that here. Also, let's fix typos.
